packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -97,241 +96,253 @@ export const RestrictedApplication = props => (
-                                <React.Fragment>
-                                  <ReconfigureFlopFlip
-                                    user={getFlopflipReconfiguration(
-                                      routerProps.match.params.projectKey
-                                    )}
-                                  />
-                                  <ProjectContainer
-                                    user={user}
-                                    match={routerProps.match}
-                                    location={routerProps.location}
-                                    // This effectively renders the
-                                    // children, which is the application
-                                    // specific part
-                                    render={props.render}
+                                    to={joinPaths(match.url, 'dashboard')}
                                   />
-                                </React.Fragment>
-                              )}
-                            />
-                          </Switch>
+                                )}
+                              />
+                              <Route
+                                exact={true}
+                                path="/"
+                                render={() =>
+                                  user ? (
+                                    // This is the only case where we need to look into localStorage
+                                    // to attempt to get the previously known `projectKey`.
+                                    // If none is found, we use the `defaultProjectKey` set by the API.
+                                    <Redirect
+                                      to={`/${selectProjectKeyFromLocalStorage() ||
+                                        user.defaultProjectKey}`}
+                                    />
+                                  ) : (
+                                    <ApplicationLoader />
+                                  )
+                                }
+                              />
+                              <Route
+                                exact={false}
+                                path="/:projectKey"
+                                render={routerProps => (
+                                  <React.Fragment>
+                                    <ReconfigureFlopFlip
+                                      user={getFlopflipReconfiguration(
+                                        routerProps.match.params.projectKey
+                                      )}
+                                    />
+                                    <ProjectContainer
+                                      user={user}
+                                      match={routerProps.match}
+                                      location={routerProps.location}
+                                      environment={props.environment}
```

Comment:
We pass the `environment` here so that we can pass it to the provider rendered within the `<ProjectContainer>`